presto-main/src/main/java/com/facebook/presto/catalogserver/RemoteMetadataManager.java

```diff
@@ -183,14 +181,15 @@ public Optional<TableHandle> getTableHandle(QualifiedObjectName tableName)
             }
 
             @Override
-            public Optional<List<ColumnMetadata>> getColumns(QualifiedObjectName tableName)
+            public List<ColumnMetadata> getColumns(TableHandle tableHandle)
             {
-                Optional<TableHandle> tableHandle = getTableHandle(tableName);
-                if (!tableHandle.isPresent()) {
-                    throw new SemanticException(MISSING_TABLE, "Table does not exist: " + tableName.toString());
-                }
-                TableMetadata tableMetadata = getTableMetadata(session, tableHandle.get());
-                return Optional.of(tableMetadata.getColumns());
+                return getTableMetadata(session, tableHandle).getColumns();
+            }
+
+            @Override
+            public Map<String, ColumnHandle> getColumnHandles(TableHandle tableHandle)
+            {
+                return RemoteMetadataManager.this.getColumnHandles(session, tableHandle);
```

Comment:
what's with the prefix here? just `getColumnHandles(session, tableHandle)` should do the trick.

Reply:
I think the function overloading does not work with inner classes. So this needs the qualifier with outer class to resolve the function.